### DISCUSSION TITLE
Yocto build issue on aarch64 + Ubuntu20.04

### DISCUSSION THREAD

**Discussion by JunWu-ARM** [ncroxon/gnu-efi#53] [Q&A] [answered]
Hi All:

I met with a build error, found it when ycoto build gnu-efi-native in an aarch64 host with Ubuntu 20.04 which has local installed binutils v2.34. As a part of binutils, objcopy before v2.38 can not support --target efi-app-aarch64 and the build will fail. The build error only happens in sub directory apps. My understanding is the apps folder include mainly the example applications.

I am trying to propose a fix to avoid build sub directory apps (in a yocto native build) in the makefile like the following patch. It works the same for cross compiling.
 
wondering if this fix is OK or what is the best approach?

@ncroxon 

Many thanks
Jun
 
---
 Makefile | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e58d02a..978ad05 100644
--- a/Makefile
+++ b/Makefile
@@ -43,7 +43,12 @@ VPATH = $(SRCDIR)
 
 include $(SRCDIR)/Make.defaults
 
-SUBDIRS = lib gnuefi inc apps
+SUBDIRS = lib gnuefi inc
+
+# build apps as example when cross compile
+ifneq ($(CROSS_COMPILE),)
+SUBDIRS += apps
+endif
 
 all:	check_gcc $(SUBDIRS)
 
-- 
2.34.1

**Comment by gmbr3** [comment, marked as answer]
Yocto is outdated - 3.0.18
https://github.com/yoctoproject/poky/commits/master/meta/recipes-bsp/gnu-efi
Update to 4.0.0

**Reply by JunWu-ARM** [reply]
Thanks for the answer. Before taking the 4.0.0 version, is it possible we can have a patch to fix this issue in scarthgap branch https://layers.openembedded.org/layerindex/recipe/398250

**Reply by gmbr3** [reply]
> Thanks for the answer. Before taking the 4.0.0 version, is it possible we can have a patch to fix this issue in scarthgap branch https://layers.openembedded.org/layerindex/recipe/398250

Just run make gnuefi instead
I don't have access to yocto to actually fix it or update to 4.0.0 - that'd be @rpurdie I believe

**Reply by JunWu-ARM** [reply]
> Just run make gnuefi instead
Thanks for confirmation. It makes sense before we take 4.0.0.

**Reply by rossburton** [reply]
Yocto wouldn't take an upgrade in a stable branch unless there's a good reason (security fixes, etc).  Can you point to a specific change in 4.0.0 which resolves this issue that can be backported?

**Reply by rpurdie** [reply]
I'm going to defer to @rossburton on this one as he is also at ARM. FWIW we don't take upgrades in stable branches without good reasons. Patches to change the build target is probably the way to go, we don't like building things we don't need/use!
